cms/models.py

```diff
@@ -155,13 +155,20 @@ def course_certificate(
 
         # Get a CertificatePage to serve this request
         certificate_page = (
-            certificate.course_run.course.page.certificate_page
-            if certificate.course_run.course.page
-            else None
+            certificate.certificate_page_revision.as_page_object()
+            if certificate.certificate_page_revision
+            else (
+                certificate.course_run.course.page.certificate_page
+                if certificate.course_run.course.page
+                else None
+            )
         )
+
         if not certificate_page:
             raise Http404()
 
+        if not certificate.certificate_page_revision:
+            certificate.save()
```

Comment:
This is fine too but I was thinking of something that might serve us better (This might be even a better use case in xPRO where we have existing certificates in production, in MITxOnline we only have certificates in RC):

How about we create a data migration while we are moving towards the revision-based certificates, e.g. when [this migration](https://github.com/mitodl/mitxonline/pull/903/files#diff-1f2100c0ef3efdd82b047e419dc49af4b0f64c985a6be6e6610a8cf4fe79851aR1) runs, we migrate the existing certificates in the database to use the latest revision of the certificate for the specific course runs present at that moment. 

The benefit is that we will have consistent data right from the start, we won't need to call the save method here every time the revision isn't found. We might even want to set [this field](https://github.com/mitodl/mitxonline/pull/903/files#diff-d5de5fce23898af38cf96ba73fbdae70cb947ffae1e47551a73f5fccbabadaf9R558-R560) as non-nullable.

Before you implement this, Let's here @rhysyngsun 's thoughts on this too?

Reply:
I'm fine with enforcing it with a migration instead of an extra write.